### PR TITLE
new IAM role for jenkins

### PR DESCRIPTION
**Pull request by mattops**

### CODE REVIEW

File: azure/jenkins-dev-test-role.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "IsCustom": true,
+  "Description": "All permissions needed for jenkins jobs",
+  "Actions": [
+    "Microsoft.Web/certificates/*",
+    "Microsoft.Network/dnsZones/*/read",
+    "Microsoft.Network/dnsZones/TXT/write",
+    "Microsoft.Network/dnsZones/TXT/delete",
+    "Microsoft.Resources/deployments/*",
+    "Microsoft.Keyvault/vaults/*",
+    "Microsoft.Web/serverfarms/*"
```
Comment (glenjamin) [inline azure/jenkins-dev-test-role.json]: Some of these are a little broad, could we trim down vaults and serverfarms before rolling out to prod?

Comment (mattops) [reply azure/jenkins-dev-test-role.json]: I've tried setting lesser permissions but it doesn't work.  I dont understand exactly why as the ARM should only need to read the vault.  Similar for the serverfarm.

Comment (mattops) [reply azure/jenkins-dev-test-role.json]: Here's the ARM template https://github.com/ministryofjustice/digital-studio-infra/blob/master/shared/ssl_cert_only.template.json

File: azure/jenkins-dev-test-role.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "IsCustom": true,
+  "Description": "All permissions needed for jenkins jobs",
+  "Actions": [
+    "Microsoft.Web/certificates/*",
+    "Microsoft.Network/dnsZones/*/read",
+    "Microsoft.Network/dnsZones/TXT/write",
+    "Microsoft.Network/dnsZones/TXT/delete",
+    "Microsoft.Resources/deployments/*",
+    "Microsoft.KeyVault/vaults/*",
+    "Microsoft.Web/serverfarms/*"
```
Comment (glenjamin) [inline azure/jenkins-dev-test-role.json]: Happy to merge this as-is, but i'd like to limit serverfarms a bit more if possible

Comment (mattops) [reply azure/jenkins-dev-test-role.json]: Ticket raised with Azure to investigate further limits. Merger code for now.

**Pull request merged**